operators/endpointmetrics/controllers/observabilityendpoint/observabilityaddon_controller.go

```diff
@@ -153,6 +159,7 @@ func (r *ObservabilityAddonReconciler) Reconcile(ctx context.Context, req ctrl.R
 		hubSecret,
 	)
 	if err != nil {
+		log.Error(err, "Failed to get hub info secret")
```

Comment:
should we log the error here?

Reply:
same comment throughout actually to be honest. If we don't intend to log the message as to why we failed perhaps we can just wrap it and return the errors and log at the top level?

Reply:
Yes good point. I was considering the reconcile function as the top level one. But errors are indeed logged by the controller runtime. Removing those.